src/pkgcore/ebuild/repo_objs.py

```diff
@@ -170,6 +171,216 @@ def __init__(self, metadata_xml, manifest):
         self.manifest = manifest
 
 
+class ProjectMember(object):
+    """Data on a single project member.
+
+    :type email: C{unicode} object
+    :ivar email: email address.
+    :type name: C{unicode} object or C{None}
+    :ivar name: full name
+    :type role: C{unicode} object or C{None}
+    :ivar role: role within the project.
+    :type is_lead: C{bool}
+    :ivar is_lead: whether the member is a project lead.
+    """
+
+    __slots__ = ('email', 'name', 'role', 'is_lead')
+
+    def __init__(self, email, name=None, role=None, is_lead=None):
+        if email is None:
+            raise ValueError('email for project member must not be null')
+        self.email = email
+        self.name = name
+        self.role = role
+        self.is_lead = is_lead
+
+    def __eq__(self, other):
```

Comment:
FTR, I've tried using `klass.generic_compare` but it didn't work. Does it work anymore with py3?

Reply:
`s/generic_compare/generic_equality/`

Reply:
I think it still works, how were you using it or what issues were you running into? It needs to be set as a metaclass and then you need to define __attr_comparison__ properly.

Reply:
```
    __metaclass__ = klass.generic_equality
    __attr_comparison__ = ('email', 'name', 'role', 'is_lead')
```

and the tests fail comparison, while they work with explicit `__eq__`.

Reply:
py3 changed how you specify metaclasses in [PEP 3115](https://www.python.org/dev/peps/pep-3115/). What happens if you switch to that format?

Reply:
Yep, that works.